kuma/users/views.py

```diff
@@ -477,11 +477,19 @@ def get_form(self, form_class=None):
         self.email_addresses = {}
         form = super(SignupView, self).get_form(form_class)
         form.fields['email'].label = _('Email address')
+
+        # When no username is provided, default to the local-part of the email address
+        if form.initial.get('username', '') == '':
+            email = form.initial.get('email', '')
+            if isinstance(email, tuple):
```

Comment:
I'm not sure if it helps but you can do:
```suggestion
            if isinstance(email, (tuple, list)):
```
which is useful sometimes when you're not entirely sure what type it's going to be but it's some iterable.